clippy_lints/src/cognitive_complexity.rs

```diff
@@ -112,8 +112,7 @@ impl<'tcx> Visitor<'tcx> for CCHelper {
         walk_expr(self, e);
         match e.node {
             ExprKind::Match(_, ref arms, _) => {
-                let arms_n: u64 = arms.iter().map(|arm| arm.pats.len() as u64).sum();
-                if arms_n > 1 {
+                if arms.len() > 1 {
```

Comment:
If I'm reading the intent of this code you might want to recurse and count each or-pattern as `cc += 1` but I think the change you made makes more sense.

Reply:
yeah i debated this but i think we might need a more careful CC accounting for pats here